test/distributed/_shard/test_replicated_tensor.py

```diff
@@ -197,3 +198,89 @@ def test_replicated_tensor_inter_op_sharded_tensor_errors(self):
 
         with self.assertRaisesRegex(RuntimeError, 'not supported for ShardedTensor'):
             st1 % replica_tensor
+
+    @with_comms(init_rpc=False)
+    @skip_if_lt_x_gpu(4)
+    @requires_nccl()
+    def test_with_ddp(self):
+        # Test Replicated params for DDP
+        replica_tensor = ReplicatedTensor(torch.rand(4, 8, device=self.rank))
+        model = torch.nn.Linear(8, 2).cuda(self.rank)
+        optim = torch.optim.SGD(model.parameters(), lr=0.1)
+        ddp = DDP(model)
+
+        # Test module.parameters.
+        params = list(ddp.parameters())
+        self.assertEqual(2, len(params))
+        self.assertEqual(ddp.module.weight, params[0])
+        self.assertEqual(ddp.module.bias, params[1])
+
+        params = list(model.parameters())
+        self.assertEqual(2, len(params))
+        self.assertEqual(model.weight, params[0])
+        self.assertEqual(model.bias, params[1])
+
+        # Validate output
+        out = ddp(replica_tensor)
```

Comment:
Should we test parity with replicated tensor vs non replicated tensor?

Reply:
So the existing DDP tests already do this where they end up running DDP with replicated tensor and compare with local model run.